revscoring/utilities/tests/test_union_merge_observations.py

```diff
@@ -13,14 +14,19 @@ def test_union():
         data_dir + "/labeled_foo.json",
     ]
 
-    # Receive the output as a string buffer.
-    out_file = io.StringIO()
+    # Receive the output in a temporary file.
+    (fh, out_file) = tempfile.mkstemp()
 
     # Do the union.
-    DataUnion().union(in_files, out_file)
+    argv = in_files + ["--output", out_file]
+    revscoring.utilities.union_merge_observations.main(argv)
```

Comment:
There's so much logic in main() that I'd rather test that.  I'm especially wary of the big read_observations mush.